src/components/InsertComment.jsx

```diff
@@ -0,0 +1,59 @@
+import { postComment } from "../api";
+import { useEffect, useState } from 'react';
+
+
+
+
+export const InsertComment = ({review_id, setComments}) => {
+    const [newComment, setNewComment] = useState("");
+    const [isLoading, setIsLoading] = useState(false);
+    
+    const handleSubmit = (e) => {
+        e.preventDefault()
+        setIsLoading(true)
+
+
+        if (newComment !== "") {
+            setComments((currComments) => {
+                const updatedComments = [{
+                    author: "happyamy2016",
+                    body: newComment,
+                    created_at: new Date(Date.now()).toString(),
+                    votes: 0,
+                    comment_id: Date.now()
+                }, ...currComments]
+                return updatedComments
+            })
+            postComment(review_id, newComment)
+                .then(() => { 
+                    setIsLoading(false);
+                })
+                .catch(() => { setComments((currComments) => {
+                    const updatedComments = [currComments]
+                    alert('Your comment did not go through, please try again.')
+                    return updatedComments.shift()
+                })
+                    })
+                setNewComment("")
+        } else {
+            alert("You cannot add an empty comment...");
+            setIsLoading(false)
+        }
+        
+    }
+    
+
+    return isLoading ? <section> ...sending comment </section> :
+        <section className="Form">
+            <br/>
+            <form onSubmit={handleSubmit}>
+                <label> <strong>Add comment</strong> </label>
+                <textarea 
+                    id="newComment"
+                    value={newComment}
+                    onChange={(e) => setNewComment(e.target.value)}/> <br/>
```

Comment:
Good decision to use a `textarea` - you may want to add the` 'required' `tag in to ensure the form cannot be submitted if this isn't filled in. I can see deal with this in code above using an alert()... the alert() function doesn't always work nicely on mobiles